healthtip/web.go

```diff
@@ -47,6 +53,45 @@ func Run(dbCon *sql.DB) error {
 	return nil
 }
 
+func authMiddleware(next func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sharedKeyPath := os.Getenv("AUTH0_PUBLIC_KEY_PATH")
+		audience := []string{os.Getenv("AUTH0_AUDIENCE")}
+		sharedKey, err := ioutil.ReadFile(sharedKeyPath)
+		if err != nil {
+			log.Println("PEM file at " + sharedKeyPath + " could not be read")
+			handleError(w, r, http.StatusInternalServerError, "Internal error")
+			return
+		}
+		secret, err := joseutil.LoadPublicKey(sharedKey)
+		if err != nil {
+			log.Println("Could not load public key from PEM file.")
+			handleError(w, r, http.StatusInternalServerError, "Internal error")
+			return
+		}
+		secretProvider := auth0.NewKeyProvider(secret)
+		configuration := auth0.NewConfiguration(secretProvider, audience, "https://"+os.Getenv("AUTH0_DOMAIN")+".auth0.com/", jose.RS256)
+		validator := auth0.NewValidator(configuration, nil)
+
+		token, err := validator.ValidateRequest(r)
+
+		// Determine user Id.
+		claims := jwt.Claims{}
+		token.Claims(secret, &claims)
+		userId := strings.Split(claims.Subject, "|")[1]
+		ctx := context.WithValue(r.Context(), "userId", userId)
+
+		if err != nil {
+			log.Println(err)
```

Comment:
I think this can probably use our helper method handleError(w, r, http.StatusUnauthorized, "Unauthorized").